src/App.svelte

```diff
@@ -82,21 +92,65 @@
-        <Registrations {wallet} session={$session} />
-        <Seeds {wallet} session={$session} />
-        <Faucet {wallet} />
-        <Route path="/:addressOrName" let:params>
-          <Profile addressOrName={params.addressOrName} {wallet} />
-        </Route>
-        <Projects {wallet} />
-      </Router>
+      {#if $activeRouteStore.type === "home"}
+        <Home />
+      {:else if $activeRouteStore.type === "faucet"}
+        {#if $activeRouteStore.params.activeView.type === "form"}
+          <FaucetForm {wallet} />
+        {:else if $activeRouteStore.params.activeView.type === "withdraw"}
+          <FaucetWithdraw
+            {wallet}
+            amount={$activeRouteStore.params.activeView.params.amount} />
+        {/if}
+      {:else if $activeRouteStore.type === "seeds"}
+        <Seeds
+          {wallet}
+          session={$session}
+          host={$activeRouteStore.params.host} />
+      {:else if $activeRouteStore.type === "registration"}
+        {#if $activeRouteStore.params.activeView.type === "validateName"}
+          <RegistrationValidateName {wallet} />
+        {:else if $activeRouteStore.params.activeView.type === "checkNameAvailability"}
+          <RegistrationCheckNameAvailability
+            {wallet}
+            name={$activeRouteStore.params.activeView.params.nameOrDomain}
+            owner={$activeRouteStore.params.activeView.params.owner} />
+        {:else if $activeRouteStore.params.activeView.type === "register"}
+          {#if $session}
+            <RegistrationRegister
+              {wallet}
+              name={$activeRouteStore.params.activeView.params.nameOrDomain}
+              owner={$activeRouteStore.params.activeView.params.owner}
+              session={$session} />
+          {:else}
+            <ErrorModal
+              message={"You must connect your wallet to register"}
+              on:close={() => navigate("/registration")} />
+          {/if}
+        {:else if $activeRouteStore.params.activeView.type === "view"}
+          <RegistrationView
+            {wallet}
+            retry={$activeRouteStore.params.activeView.params.retry}
+            domain={$activeRouteStore.params.activeView.params.nameOrDomain} />
+        {:else}
+          {unreachable($activeRouteStore.params.activeView)}
+        {/if}
+      {:else if $activeRouteStore.type === "vesting"}
+        <Vesting {wallet} session={$session} />
+      {:else if $activeRouteStore.type === "projects"}
+        <Projects
+          {wallet}
+          profile={$activeRouteStore.params.profile || null}
+          seed={$activeRouteStore.params.seed || null}
+          peer={$activeRouteStore.params.peer || null}
+          urn={$activeRouteStore.params.urn} />
+      {:else if $activeRouteStore.type === "profile"}
+        <Profile
+          addressOrName={$activeRouteStore.params.addressOrName}
+          {wallet} />
+      {:else}
+        <NotFound title="404" subtitle="Nothing here" />
+      {/if}
```

Comment:
Can we create a `Route` component? This is pretty unreadable..

Reply:
I'd rather not add another layer of indirection, if possible, because it makes it harder to work with in my opinion. We could however push some of the screens that have sub-routing back down into a component as a compromise to do away with the sub-branching.

Reply:
>  push some of the screens that have sub-routing back down into a component

Will provide a commit for that, so we can see if it makes it more readable

Reply:
7ebc71d6ac21605c09821e3661f4f571b4d040a1